lib/doorkeeper/oauth/refresh_token_request.rb

```diff
@@ -51,12 +53,19 @@ def create_access_token
           client
         )
 
-        @access_token = AccessToken.create!(
+        create_params = {
           application_id: refresh_token.application_id,
+          expires_in: expires_in,
           resource_owner_id: refresh_token.resource_owner_id,
           scopes: scopes.to_s,
-          expires_in: expires_in,
-          use_refresh_token: true)
+          use_refresh_token: true
+        }
+
+        unless server.refresh_token_revoked_in.zero?
+          create_params[:previous_refresh_token] = refresh_token.refresh_token
+        end
```

Comment:
Why does this need to be surrounded by the conditional? Can't we/shouldn't be always store the previous refresh token?